### ISSUE TITLE
[Q] Visualize the historical change of the bar chart during the training.

### ISSUE THREAD

**Issue by Sun-Rider** [wandb/client#2407]
Hi,

I'm new to the wandb, and it is probably a basic question. But I didn't find any solution of it.

I've already have the bar-chart, and I'm wondering is there any way to visualize the change during the training (as I upload the chart based on a certain step).

![Screenshot from 2021-07-17 11-27-58](https://user-images.githubusercontent.com/33716428/126021954-17ef5127-6db5-467a-9079-560e49a7e635.png)

Cheers

**Comment by vanpelt** [comment]
Hey @Sun-Rider can you share a link to the run in question?  You should be able to click the "gear" icon in the upper left and slide to earlier steps in the training process.  Also, calling `wandb.log({"loss": X})` as your model trains will log many points that you can use to render line plots of progress during the corse of training.

**Comment by pyedog1976** [comment]
Hi,

The run's link is [here](https://wandb.ai/pyedog1976/testing-bar-chart?workspace=user-pyedog1976). 
The code I used is something like
``` python
        total_number = len(targets_prob.flatten(0))
        boundary = ["< 0.1", "0.1~0.2", "0.2~0.3",
                    "0.3~0.4", "0.4~0.5", "0.5~0.6",
                    "0.6~0.7", "0.7~0.8", "0.8~0.9",
                    "> 0.9"]
        rate = [torch.sum((torch.logical_and((i - 1) / 10 < targets_prob, targets_prob < i / 10))==True)
                / total_number for i in range(1, 11)]
        pass_rate = [[label, val] for (label, val) in zip(boundary, rate)]
```
with upload, 
``` python
wandb.log({"xxx" : wandb.plot.bar(xxx, "xxx", "xxx",
                               title="xxx")
```

**Comment by Sun-Rider** [comment]
Hi, @vanpelt

The run's link is here.
The code I used is something like

        total_number = len(targets_prob.flatten(0))
        boundary = ["< 0.1", "0.1~0.2", "0.2~0.3",
                    "0.3~0.4", "0.4~0.5", "0.5~0.6",
                    "0.6~0.7", "0.7~0.8", "0.8~0.9",
                    "> 0.9"]
        rate = [torch.sum((torch.logical_and((i - 1) / 10 < targets_prob, targets_prob < i / 10))==True)
                / total_number for i in range(1, 11)]
        pass_rate = [[label, val] for (label, val) in zip(boundary, rate)]
with upload,

wandb.log({"xxx" : wandb.plot.bar(xxx, "xxx", "xxx",
                               title="xxx")

**Comment by vanpelt** [comment]
Hey @Sun-Rider I'm asking our internal team what they think the best solution for this is. 

Until we hear back, my first thought is that this is most similar to our `wandb.Histogram` [native logger](https://docs.wandb.ai/ref/python/data-types/histogram#examples).  You would likely want to construct your own `np.histogram` with the bins configured to be your boundaries.  This allows you to visually see the distributions changing over time:

![image](https://user-images.githubusercontent.com/17/126341287-2129bdc6-61ec-44d5-bf79-d32aace24c95.png)

**Comment by vanpelt** [comment]
@Sun-Rider I heard back from the team.  You can add a slider to your bar chart by following theses steps:

1. Click the pencil icon on your chart to edit the definition.  Then select "summaryTable" in the query section and change it to "historyTable".
![image](https://user-images.githubusercontent.com/17/126358119-13119e9b-6540-4f4b-b297-5d29c2bd5cc6.png)
2. Click the "Edit" button next to "Bar Chart"
![image](https://user-images.githubusercontent.com/17/126358327-f11f2e8f-a200-4b2f-89ec-8ddbcd24fa7d.png)
3. Choose "OTHER SETTINGS" from the chart editor and select the "Show step selector" option.
![image](https://user-images.githubusercontent.com/17/126358281-29011307-1d44-49c1-8d63-d6a4b09c7844.png)
4. Now your chart will have a gear icon you can select to change the step
![image](https://user-images.githubusercontent.com/17/126358484-36912635-03e8-46b9-b621-aab128f73a1c.png)

**Comment by Sun-Rider** [comment]
Very useful!
Thank you so much!

**Comment by jayelm** [comment]
This is super useful, and would love to see this in the documentation e.g. as a FAQ! I imagine it's a fairly common use case to want to visualize how custom charts progress over time.
